diff --git a/libs/androidfw/ApkParsing.cpp b/libs/androidfw/ApkParsing.cpp
index 1234567..abcdefg 100644
--- a/libs/androidfw/ApkParsing.cpp
+++ b/libs/androidfw/ApkParsing.cpp
@@ -17,6 +17,7 @@
 #include "androidfw/ApkParsing.h"
 #include <algorithm>
 #include <array>
+#include <cstring>
 #include <stdlib.h>
 #include <string_view>
 #include <sys/types.h>
diff --git a/libs/androidfw/ResourceTypes.cpp b/libs/androidfw/ResourceTypes.cpp
index f12966341..6cc1a86d4 100644
--- a/libs/androidfw/ResourceTypes.cpp
+++ b/libs/androidfw/ResourceTypes.cpp
@@ -45,7 +45,7 @@
 #include <utils/String16.h>
 #include <utils/String8.h>
 #include <android-base/logging.h>
-#ifdef __ANDROID__
+#if 0
 #include <binder/TextOutput.h>
 
 #endif
diff --git a/libs/androidfw/include/androidfw/ResourceTypes.h b/libs/androidfw/include/androidfw/ResourceTypes.h
index cb47034ee..e5cfea1d7 100644
--- a/libs/androidfw/include/androidfw/ResourceTypes.h
+++ b/libs/androidfw/include/androidfw/ResourceTypes.h
@@ -519,7 +519,7 @@ struct ResStringPool_span
 
     // The range of characters in the string that this span applies to.
     uint32_t firstChar, lastChar;
-};
+} __attribute__((__packed__, aligned(1)));
 
 /**
  * Convenience class for accessing data in a ResStringPool resource.
