diff --git a/libcutils/include/cutils/trace.h b/libcutils/include/cutils/trace.h
index 7f57637..4bc2f01 100644
--- a/libcutils/include/cutils/trace.h
+++ b/libcutils/include/cutils/trace.h
@@ -19,6 +19,10 @@
 
 #include <inttypes.h>
 #include <stdatomic.h>
+#ifdef __cplusplus
+#include <atomic>
+using std::atomic_bool;
+#endif
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdio.h>
diff --git a/libpackagelistparser/packagelistparser.cpp b/libpackagelistparser/packagelistparser.cpp
index 5517b68..39785ab 100644
--- a/libpackagelistparser/packagelistparser.cpp
+++ b/libpackagelistparser/packagelistparser.cpp
@@ -23,7 +23,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/limits.h>
+#include <limits.h>
 
 #include <memory>
 
diff --git a/libutils/Threads.cpp b/libutils/Threads.cpp
index 111d46a..f50d571 100644
--- a/libutils/Threads.cpp
+++ b/libutils/Threads.cpp
@@ -91,11 +91,11 @@ struct thread_data_t {
 // Adapted from bionic's implmenetation of trampoline to make C11 thrd_create
 // work with pthread_create.
 struct libutil_thread_data {
-  android_pthread_entry _Nonnull entry_func;
-  void* _Nullable entry_func_arg;
+  android_pthread_entry entry_func;
+  void* entry_func_arg;
 };
 
-static void* _Nonnull libutil_thread_trampoline(void* _Nonnull arg) {
+static void* libutil_thread_trampoline(void* arg) {
   libutil_thread_data *data_ptr = static_cast<libutil_thread_data*>(arg);
   int result = data_ptr->entry_func(data_ptr->entry_func_arg);
   delete data_ptr;
