diff --git a/libsepol/src/private.h b/libsepol/src/private.h
index 1500bbc..f3ce715 100644
--- a/libsepol/src/private.h
+++ b/libsepol/src/private.h
@@ -16,8 +16,8 @@
 #include <errno.h>
 
 #ifdef __APPLE__
-#define __BYTE_ORDER  BYTE_ORDER
-#define __LITTLE_ENDIAN  LITTLE_ENDIAN
+#define __BYTE_ORDER BYTE_ORDER
+#define __LITTLE_ENDIAN LITTLE_ENDIAN
 #endif
 
 #if __BYTE_ORDER == __LITTLE_ENDIAN
@@ -37,12 +37,12 @@
 #endif
 
 #undef min
-#define min(a,b) (((a) < (b)) ? (a) : (b))
+#define min(a, b) (((a) < (b)) ? (a) : (b))
 
 #undef max
-#define max(a,b) ((a) >= (b) ? (a) : (b))
+#define max(a, b) ((a) >= (b) ? (a) : (b))
 
-#define ARRAY_SIZE(x) (sizeof(x)/sizeof((x)[0]))
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
 
 static inline int exceeds_available_bytes(const struct policy_file *fp, size_t x, size_t req_elem_size)
 {
@@ -67,9 +67,9 @@ static inline int exceeds_available_bytes(const struct policy_file *fp, size_t x
 /* Use to ignore intentional unsigned under- and overflows while running under UBSAN. */
 #if defined(__clang__) && defined(__clang_major__) && (__clang_major__ >= 4)
 #if (__clang_major__ >= 12)
-#define ignore_unsigned_overflow_        __attribute__((no_sanitize("unsigned-integer-overflow", "unsigned-shift-base")))
+#define ignore_unsigned_overflow_ __attribute__((no_sanitize("unsigned-integer-overflow", "unsigned-shift-base")))
 #else
-#define ignore_unsigned_overflow_        __attribute__((no_sanitize("unsigned-integer-overflow")))
+#define ignore_unsigned_overflow_ __attribute__((no_sanitize("unsigned-integer-overflow")))
 #endif
 #else
 #define ignore_unsigned_overflow_
@@ -94,13 +94,3 @@ extern size_t put_entry(const void *ptr, size_t size, size_t n,
 		        struct policy_file *fp);
 extern int str_read(char **strp, struct policy_file *fp, size_t len);
 
-#ifndef HAVE_REALLOCARRAY
-static inline void* reallocarray(void *ptr, size_t nmemb, size_t size) {
-	if (size && nmemb > (size_t)-1 / size) {
-		errno = ENOMEM;
-		return NULL;
-	}
-
-	return realloc(ptr, nmemb * size);
-}
-#endif
